stacks_queues/queue.py

```diff
@@ -15,47 +15,87 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
-        """ Adds an element to the Queue
-            Raises a QueueFullException if all elements
-            In the store are occupied
-            returns None
-        """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException('Queue is full')
+
+        if self.size == 0:
+            self.front = 0
+            self.rear = 0
+        
+        self.store[self.rear] = element
+        #when we enqueue we have to move the rear
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        #check if empty, raise exception
+        #find and store the front *element* in a temp variable
+        #move front to the next index
+        #return old front element
+
+        if self.empty():
+            raise QueueEmptyException('Queue is empty')
+
+        temp = self.store[self.front]
+        self.store[self.front] = None
+        #when we dequeue we have to move the front
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+
+        return temp
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.size == 0:
+            return None
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        return self.size == 0
 
     def __str__(self):
+
         """ Returns the Queue in String form like:
             [3, 4, 7]
             Starting with the front of the Queue and
             ending with the rear of the Queue.
         """
-        pass
+        # int_list = self.store[:]
+        # int_list = list(filter(None, int_list))
+        # string_list = [str(int) for int in int_list]
+        # my_string = ', '.join(string_list)
+        # result = '[' + my_string + ']'
+        # return result
+        
+        #first collect elements of the queue in a list
+        int_list = []
+        for i in range(self.size):
+            #because of the circular buffer the index needs to "revolve":
+            current_i = (self.front + i) % self.buffer_size
+            #convert the int to a string
+            int_list.append(str(self.store[current_i]))
+
+        #return a string resembling a list
+        return '[' + ', '.join(int_list) + ']'
```

Comment:
This does the same thing!
```suggestion
        return str(int_list)
```